helix-vcs/src/git.rs

```diff
@@ -76,29 +78,21 @@ impl DiffProvider for Git {
         let file_oid = find_file_in_commit(&repo, &head, file)?;
 
         let file_object = repo.find_object(file_oid)?;
-        let mut data = file_object.detach().data;
-        // convert LF to CRLF if configured to avoid showing every line as changed
-        if repo
-            .config_snapshot()
-            .boolean("core.autocrlf")
-            .unwrap_or(false)
-        {
-            let mut normalized_file = Vec::with_capacity(data.len());
-            let mut at_cr = false;
-            for &byte in &data {
-                if byte == b'\n' {
-                    // if this is a LF instead of a CRLF (last byte was not a CR)
-                    // insert a new CR to generate a CRLF
-                    if !at_cr {
-                        normalized_file.push(b'\r');
-                    }
-                }
-                at_cr = byte == b'\r';
-                normalized_file.push(byte)
-            }
-            data = normalized_file
+        let data = file_object.detach().data;
+        // Get the actual data that git would make out of the git object.
+        // This will apply the user's git config or attributes like crlf conversions.
+        if let Some(work_dir) = repo.work_dir() {
+            let rela_path = file.strip_prefix(work_dir)?;
+            let rela_path = gix::path::try_into_bstr(rela_path)?;
+            let (mut pipeline, _) = repo.filter_pipeline(None)?;
```

Comment:
Just as a note: It's a bit sad that the filter pipeline can't be reused under the circumstances. if this should ever be an issue that can be achieved by dropping down to the [internal plumbing types](https://github.com/Byron/gitoxide/blob/17638628586900d43d730e6ed2a0862d8e408f29/gix/src/filter.rs#L70-L72).

However, I wouldn't go there as this will seriously complicate the code - I just wanted you to know that you can to shave off some cycles.